packages/react-components/react-drawer/src/components/Drawer/useDrawer.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { getNativeElementProps, useControllableState } from '@fluentui/react-utilities';
+import { Dialog, DialogProps, DialogSurface, DialogSurfaceProps } from '@fluentui/react-dialog';
+
+import type { DrawerProps, DrawerState } from './Drawer.types';
+
+/**
+ * @internal
+ * Create the state required to render DrawerDialog.
+ * @param props - props from this instance of Drawer
+ */
+const useDrawerDialog = ({ size, open, onOpenChange, modal, children }: DrawerProps) => {
+  const dialog = React.useMemo<DialogProps>(() => {
+    return {
+      open,
+      onOpenChange,
+      modalType: modal ? 'modal' : 'non-modal',
+      children: <></>,
```

Comment:
this looks fishy / from code maintenance POV  one might get confused what is purpose of this? why it cannot be `null` ?

Reply:
The Dialog children cannot be null. Dialog is renderless component, that expect children elements to be passed.
Due to the way our Slot system works, at this point, the component is only creating the props that will later be passed to our slots.

Reply:
It would work and we wouldn't need this file to be .tsx.
I will try to rewrite this to at least override the types, so we wouldn't need this Fragment.

Reply:
> It would work and we wouldn't need this file to be .tsx.

you can always use direct api if needed -> `React.createElement(React.Fragment)`

Reply:
This is now fixed.